File: HTTPServer.js
const http = require('http');
const fs = require('fs');
const path = require('path');

const server = http.createServer((req, res) => {
    if (req.method === 'GET') {
        let filePath = path.join(__dirname, req.url === '/' ? 'index.html' : req.url);
        const extname = String(path.extname(filePath)).toLowerCase();
        const mimeTypes = {
            '.html': 'text/html',
            '.js': 'text/javascript',
            '.css': 'text/css',
            '.json': 'application/json',
            '.png': 'image/png',
            '.jpg': 'image/jpg',
            '.gif': 'image/gif',
            '.svg': 'image/svg+xml',
            '.wav': 'audio/wav',
            '.mp4': 'video/mp4',
            '.woff': 'font/woff',
            '.woff2': 'font/woff2',
            '.ttf': 'font/ttf',
            '.eot': 'font/eot',
            '.otf': 'font/otf',
            '.ico': 'image/x-icon',
            '.zip': 'application/zip',
            '.pdf': 'application/pdf',
            '.doc': 'application/msword',
            '.xls': 'application/vnd.ms-excel',
            '.ppt': 'application/vnd.ms-powerpoint',
        };

        const contentType = mimeTypes[extname] || 'application/octet-stream';

        fs.readFile(filePath, (error, content) => {
            if (error) {
                if (error.code === 'ENOENT') {
                    res.writeHead(404, { 'Content-Type': 'text/html' });
                    res.end('<h1>404 Not Found</h1>', 'utf-8');
                } else {
                    res.writeHead(500);
                    res.end('Sorry, there was an error: ' + error.code + ' ..\n');
                }
            } else {
                res.writeHead(200, { 'Content-Type': contentType });
                res.end(content, 'utf-8');
            }
        });
    } else {
        res.writeHead(405, { 'Content-Type': 'text/html' });
        res.end('<h1>405 Method Not Allowed</h1>', 'utf-8');
    }
});

const PORT = process.env.PORT || 3000;
server.listen(PORT, () => {
    console.log(`Server running at http://localhost:${PORT}/`);
});

File: HTTPServer.R
library(httpuv)

# Define the server
server <- startServer("0.0.0.0", 8080, list(
  call = function(req) {
    if (req$REQUEST_METHOD == "GET") {
      # Serve static HTML files
      file_path <- paste0("www", req$PATH_INFO)
      if (file.exists(file_path)) {
        res <- list(
          status = 200,
          headers = list("Content-Type" = "text/html"),
          body = readLines(file_path)
        )
      } else {
        res <- list(
          status = 404,
          headers = list("Content-Type" = "text/plain"),
          body = "404 Not Found"
        )
      }
      return(res)
    }
  }
))

# Keep the server running
cat("Server is running at http://0.0.0.0:8080\n")
while (TRUE) {
  Sys.sleep(1)
}
